### PR TITLE
search: allow searching journal with spaces

### PR DESCRIPTION
**Pull request by nooraangelva**
Ref: #2359

### CODE REVIEW

File: backend/inspirehep/search/mappings/v7/records/hep.json [hunk 1]
```diff
@@ -1534,6 +1535,13 @@
         }
       },
       "normalizer": {
+        "journal_title_normalizer": {
+          "type": "custom",
+          "char_filter": [
+            "journal_title_punctuation",
+            "whitespace_normalization"
+          ]
```
Comment (drjova) [inline backend/inspirehep/search/mappings/v7/records/hep.json:1543]: you should include also the `lowercase` filter

File: backend/tests/integration/search/test_api.py [hunk 2]
```diff
@@ -90,6 +90,49 @@ def test_return_record_for_publication_info_search_with_journal_title_without_do
     assert 200 == response.status_code
 
 
+def test_return_record_for_journal_info_search_with_journal_title_with_dots_and_spaces(
```
Comment (drjova) [inline backend/tests/integration/search/test_api.py:93]: could we add test cases for all possible searches?

File: backend/inspirehep/records/marshmallow/literature/es.py [hunk 3]
```diff
@@ -220,3 +221,15 @@ def separate_authors_and_supervisors_and_populate_first_author(self, data):
...
         return data
+
+    def get_journal_title_variants(self, record):
+
+        publication_infos = list(
+            InspireRecord.get_linked_records_from_dict_field(record, "publication_info")
+        )
+        for publication_info in publication_infos:
+            record["journal_title_variants"].append(
+                " ".join(publication_info["journal_title"].split("."))
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/literature/es.py:232]:
```suggestion
                publication_info["journal_title"].split(".")
```

File: backend/inspirehep/records/marshmallow/literature/es.py [hunk 4]
```diff
@@ -220,3 +221,22 @@ def separate_authors_and_supervisors_and_populate_first_author(self, data):
...
+        publication_infos = record.get("publication_info", [])
+
+        result = []
+        for publication_info in publication_infos:
+            if "journal_title" in publication_info:
+
+                result.append(
+                    publication_info["journal_title"].replace(" ", "").split(".")
+                )
+        if result.len() > 0:
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/literature/es.py:239]:
```suggestion
        if result:
```

File: backend/inspirehep/records/marshmallow/literature/es.py [hunk 5]
```diff
@@ -220,3 +221,22 @@ def separate_authors_and_supervisors_and_populate_first_author(self, data):
             if data["authors"]:
                 data["first_author"] = data["authors"][0]
         return data
+
+    def get_journal_title_variants(self, record):
+
+        # import pdb
+        # pdb.set_trace()
+
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/literature/es.py:229]: Left overs 😅

File: backend/tests/integration/search/test_api.py [hunk 6]
```diff
@@ -90,6 +90,51 @@ def test_return_record_for_publication_info_search_with_journal_title_without_do
...
+    inspire_app,
+):
+
+    queries = ["Phys.Lett.B", "Phys.Lett.B"]
```
Comment (drjova) [inline backend/tests/integration/search/test_api.py:97]: Isn't it the same query? :)

**Review by MJedr — APPROVED** [review]

**Comment by drjova** [comment]
@MJedr could you please have another look?

**Review by MJedr — APPROVED** [review]

**Pull request merged**